libr/util/charset.c

```diff
@@ -113,24 +113,36 @@ R_API size_t r_charset_encode_str(RCharset *rc, ut8 *out, size_t out_len, const
 
 // assumes out is as big as in_len
 R_API size_t r_charset_decode_str(RCharset *rc, ut8 *out, size_t out_len, const ut8 *in, size_t in_len) {
-	char k[32];
 	char *o = (char*)out;
-	int i;
-	for (i = 0; i < in_len; i++) {
-		ut8 ch_in = in[i];
-
-		//zero terminate the string
-		snprintf (k, sizeof (k), "%c", ch_in);//snprintf (k, sizeof (k), "0x%02x", ch_in);
-		char *v = sdb_get (rc->db_char_to_hex, k, 0);
-		memmove(v, v+2, strlen (v));
-
-		//convert to ascii
-		char str_hx[32];
-		snprintf (str_hx, sizeof (str_hx), "%c", (char) strtol( v, 0, 16));
-		const char *ret = r_str_get_fail (str_hx, "?");
-		strcpy (o, ret);
-
-		o += strlen (o);
+	int j;
+	ut8 str[320];
+	size_t last_char_size;
+
+	size_t cur;
+	for (cur = 0; cur < in_len; ) {
+		memcpy (str, in, sizeof (str));
+		for (j =( in_len+1 ); j > 0; j--) {
+			str[j] = '\0';
+			//zero terminate the string
+			char *v = sdb_get (rc->db_char_to_hex, (char *) str+cur, 0);
+
+			if (v) {
+				memmove (v, v+2, strlen (v));
+
+				//convert to ascii
+				char str_hx[320];
```

Comment:
you dont need 320 chars to save two chars

Reply:
use 8 instead of 320

Reply:
or just do my *o = strtol.. suggestion and this is not necessary. multichar support is not yet there, so lets make the code simpler